test/unit/io/sequence_file/sequence_file_format_genbank_test.cpp

```diff
@@ -134,239 +147,42 @@ REFERENCE   1  (bases 1 to 5028)
 FEATURES             Location/Qualifiers
      source          1..5028
 )";
-    expected_ids[1] = R"(LOCUS ID2
+    ids[1] = R"(LOCUS ID2
 DEFINITION  Homo sapiens mRNA for prepro cortistatin like peptide, complete
             cds.
 ACCESSION   ID2
 )";
-    expected_ids[2] = R"(LOCUS ID3
+    ids[2] = R"(LOCUS ID3 lala
 DEFINITION  Homo sapiens mRNA for prepro cortistatin like peptide, complete
             cds.
 ACCESSION   ID3
 )";
-    do_read_test(input);
-}
-
-TEST_F(read, only_seq)
-{
-    std::stringstream istream{input};
-
-    for (unsigned i = 0; i < 3; ++i)
-    {
-        id.clear();
-        seq.clear();
-
-        EXPECT_NO_THROW( (format.read(istream, options, seq, std::ignore, std::ignore) ));
 
-        EXPECT_TRUE((ranges::equal(seq, expected_seqs[i])));
-    }
-}
-
-TEST_F(read, only_id)
-{
-    std::stringstream istream{input};
-
-    for (unsigned i = 0; i < 3; ++i)
-    {
-        id.clear();
-        seq.clear();
-
-        EXPECT_NO_THROW( (format.read(istream, options, std::ignore, id, std::ignore) ));
```

Comment:
so all this has moved into the test template. But are the semantics of the test identical. I noticed that some things are not checking throw or no_throw expressions?

Reply:
No there are not identical. They now test the same semantics but with the public interface.

Reply:
Yes, but the relevant parts that test the corresponding throw statments are covered?

Reply:
yes